### ISSUE TITLE
[Tutorial Request] Elementwise Operation

### ISSUE THREAD

**Issue by WenyinWei** [viennacl/viennacl-dev#285]
Hello Karl,

I am looking for the elementwise operation for a quick initialization of matrices on the GPU. And I found that viennacl uses the compile-time definitions to make these functions [matrix_operations.hpp, line 701-825 ](http://viennacl.sourceforge.net/doc/matrix__operations_8hpp_source.html). It is pretty concise, yeah, but it really needs a tutorial or some like that to illustrate how to use these functions. They even do not appear in the doxygen document that new comers would have no idea about how to do the function.

Best wishes,
Wenyin

**Comment by WenyinWei** [comment]
Hello, Karl, @karlrupp  I have some trouble while using element_op functions.

When I use the following command to get the `element_geq`, `_greater`, `_less` and etc, I got the result matrix_expression but I don't know how to handle it. I can't impose copy functions to fetch the result back to CPU or do other things. And I have no idea about what it would return, if 1 for geq, 0 for less or -1 for less?

**Comment by karlrupp** [comment]
Assign it to a matrix object, e.g. `A = element_geq(B, C);`

**Comment by WenyinWei** [comment]
![2020-02-26 16-28-29 Screenshot](https://user-images.githubusercontent.com/33152225/75326187-24538680-58b5-11ea-8e4f-4d7585b9cc14.png)
Yeah, I have tried, but still stuck by a bug. 

I tried to select the binary elementwise functions which can work.
`element_prod`, `_div`, `_pow` works,
`_eq`, `_neq`, `_greater`, etc. don't work.

![2020-02-26 16-42-58 的屏幕截图](https://user-images.githubusercontent.com/33152225/75327304-1bfc4b00-58b7-11ea-8070-8a6c581b665a.png)

**Comment by karlrupp** [comment]
Uh, then I guess that these other operators have not been implemented for matrices yet.

**Comment by WenyinWei** [comment]
Uh, Karl, I found that arctan function works well. Maybe that could serve as a temporary workaround, but just temp.

**Comment by WenyinWei** [comment]
Hello Karl, would you mind requesting the original maintainers to have a minor update? Though atan function serves as a kind of workaround, it sacrifices too much performance.

**Comment by WenyinWei** [comment]
![2020-03-01 16-49-14 的屏幕截图](https://user-images.githubusercontent.com/33152225/75622568-ad7ffb80-5bdc-11ea-9e4f-5a0b13bf867c.png)
Here is my test error of viennacl::linalg::element_abs on matrix.

**Comment by WenyinWei** [comment]
Hello @karlrupp , may I ask about whether viennacl is not maintained anymore and shall I turn to PETSc?

**Comment by karlrupp** [comment]
As I said, element_abs is not implemented for matrices yet. Thus, this is not a question about maintenance, but a feature request ;-)

Do you need `abs` for the full matrix, or just for a submatrix? For a full matrix you could reuse the existing vector kernels (I can show you how to do it).

**Comment by WenyinWei** [comment]
Yeah, Karl, I would appreciate it if you could show me how to do that when you are free. It would be better if the trick is generic for `geq` and `max`.
